resteasy-client-jetty/src/main/java/org/jboss/resteasy/client/jaxrs/engines/jetty/JettyClientEngine.java

```diff
@@ -248,6 +288,43 @@ private static RuntimeException clientException(Throwable ex, jakarta.ws.rs.core
         return ret;
     }
 
+    private static ClassLoader resolveClassLoader() {
+        if (System.getSecurityManager() == null) {
+            ClassLoader cl = Thread.currentThread().getContextClassLoader();
+            if (cl == null) {
+                cl = JettyClientEngine.class.getClassLoader();
+            }
+            return cl == null ? ClassLoader.getSystemClassLoader() : cl;
+        }
+        return AccessController.doPrivileged((PrivilegedAction<ClassLoader>) () -> {
+            ClassLoader cl = Thread.currentThread().getContextClassLoader();
+            if (cl == null) {
+                cl = JettyClientEngine.class.getClassLoader();
+            }
+            return cl == null ? ClassLoader.getSystemClassLoader() : cl;
+        });
+    }
+
+    private static boolean isMultipart(final ClientInvocation invocation) {
+        return MULTIPART_WILDCARD.isCompatible(invocation.getHeaders().getMediaType());
+    }
+
+    private static boolean canSetBoundary(final Object entity) {
```

Comment:
should this return something like `Optional<String>` as the boundary, provided by `MultipartOutput.getBoundary()`, so it can be set on the header instead of a random UUID on line 146? Otherwise wouldn't we end up with a different boundary in the header vs. the body?

I'm not totally sure how `EntityPart` writes the boundary. Not sure if this is in the path but I do see lists of `EntityPart`  being converted into `MultipartFormDataOutput` in `MultipartEntityPartWriter#create`. If so then the boundary uuid would get assigned as a random uuid there when `new MultipartFormDataOutput()` is called, and that would also need to match this header?

Reply:
The `AbstractMultipartWriter` [checks the header](https://github.com/resteasy/resteasy/blob/2f82f7db7df5ac5dc654d5c8785224ded854092e/providers/multipart/src/main/java/org/jboss/resteasy/plugins/providers/multipart/AbstractMultipartWriter.java#L52-L54) before it gets the boundary from the `MultipartOutput`.

The `Entitypart` does end up using the `AbstractMultipartFormDataWriter`.

IoW, the `MultipartOutput.getBoundary()` shouldn't be used at all if the header has been set.